src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -56,21 +56,53 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
             var spans = textView.Selection.GetSnapshotSpansOnBuffer(subjectBuffer);
 
             // Don't split strings if there is any actual selection.
-            if (spans.Count == 1 && spans[0].IsEmpty)
+            // We must check all spans to account for multi-carets.
+            if (spans.IsEmpty() || !spans.All(s => s.IsEmpty))
             {
-                var caret = textView.GetCaretPoint(subjectBuffer);
-                if (caret != null)
+                return false;
+            }
+
+            var caret = textView.GetCaretPoint(subjectBuffer);
+            if (caret != null)
+            {
+                // First, we need to verify that we are only working with string literals.
+                // Otherwise, let the editor handle all carets.
+                for (var spanIndex = 0; spanIndex < spans.Count; spanIndex++)
                 {
-                    // Quick check.  If the line doesn't contain a quote in it before the caret,
-                    // then no point in doing any more expensive synchronous work.
-                    var line = subjectBuffer.CurrentSnapshot.GetLineFromPosition(caret.Value);
-                    if (LineContainsQuote(line, caret.Value))
+                    var spanStart = spans[spanIndex].Start;
+                    var line = subjectBuffer.CurrentSnapshot.GetLineFromPosition(spanStart);
+                    if (!LineContainsQuote(line, spanStart))
                     {
-                        return SplitString(textView, subjectBuffer, caret.Value);
+                        return false;
                     }
                 }
-            }
 
+                // We now go through the verified string literals and split each of them.
+                for (var spanIndex = 0; spanIndex < spans.Count; spanIndex++)
+                {
+                    var spanStart = spans[spanIndex].Start;
+
+                    // Multi-caret case
+                    if (spanIndex > 0)
+                    {
+                        caret = textView.GetCaretPoint(subjectBuffer);
+                        if (caret == null)
+                        {
+                            return false;
+                        }
+
+                        // We change the span's starting point based on how much space was added from the last split.
```

Comment:
an easier way to do things would be to start from the *last* span and go *backwards*.  That way you don't need to update anything since later splits won't affect earlier caret positions.